cdap-ui/app/directives/widget-container/widget-radio-group/widget-radio-group.html

```diff
@@ -0,0 +1,57 @@
+<!--
+  Copyright © 2018 Cask Data, Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+  use this file except in compliance with the License. You may obtain a copy of
+  the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+  License for the specific language governing permissions and limitations under
+  the License.
+-->
+<div
+  ng-class="{
+    'widget-radio-group': true,
+    'widget-radio-group-inline': layout === 'inline'
+  }"
+>
+  <div ng-if="error">
+    <div class="text-danger">
+      {{error}}
+      <div ng-if="model">
+          <label>
+            <input
+              type="radio"
+              id="{{model}}"
+              value="{{model}}"
+              name="{{groupName}}"
+              ng-model="model"
+            />
+            {{model}}
+          </label>
+      </div>
+    </div>
+  </div>
+  <div
+    ng-class="{
+      radio: layout === 'block',
+      'radio-inline': layout === 'inline'
+    }"
+    ng-repeat="option in options"
```

Comment:
you don't need to filter out the `model` input from this list? it won't become doubled?

Reply:
I am not sure if I understand the question. This ng-repeat just renders radio buttons for each option. The logic to make it checked or not is determined by the ng-model attribute. Am I missing something here?

Reply:
you have one ngIf at line 25 that will render another radio button, i'm assuming that will render the selected option?

Reply:
That will get rendered only if there was no options specified as part of the widget json. If there are any error (which as of now is only non-array options) then that gets rendered.